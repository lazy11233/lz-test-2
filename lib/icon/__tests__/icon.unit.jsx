import React from 'react';
import TestRenderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import Icon from '../icon';

describe('icon', () => {
  it('should ', () => {
    const json = TestRenderer.create(<Icon name="wechat" />).toJSON();
    expect(json).toMatchSnapshot();
  });
  it('onClick', () => {
    const fn = jest.fn();
    const alipayIcon = mount(<Icon name="alipay" onClick={fn} />);
    alipayIcon.find('svg').simulate('click');
    expect(fn).toBeCalled();
  });
});