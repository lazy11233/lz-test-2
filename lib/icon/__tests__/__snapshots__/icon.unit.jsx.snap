// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon should  1`] = `
<svg
  className="lui-icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
